File: ago-dic-2022/Antonio Carbajal/Insertion_Sort.java
package Actividades;

import java.util.*;

public class Insertion_Sort {
	
	public static void main(String[]args) {
		
		int[] A= {10,1,5,16,25,87,48,65};
		System.out.println("Arreglo Original:" + Arrays.toString(A));
		
		for(int i=0; i<A.length;i++) {
			int temp=A[i];
			int j= i-1;
			while(j>=0 && temp<=A[j]) {
				A[j+1]=A[j];
				j=j-1;
			}
			A[j+1]=temp;
		}
		
		System.out.println("Arreglo Ordenado:" + Arrays.toString(A));
	}

}

File: ago-dic-2023/ErickRodriguezRamirez/algoritmos/insertion-sort.js
const arreglo = [10, 3, 7, 1, 8, 2, 4, 9, 6, 5]; // Ejemplo de un arreglo desordenado

for (var i = 1; i < arreglo.length; i++) {
  var k = arreglo[i];
  var j = i - 1;

  while (j >= 0 && arreglo[j] > k) {
    arreglo[j + 1] = arreglo[j];
    j = j - 1;
  }

  arreglo[j + 1] = k;
}

console.log(arreglo); // Imprime el arreglo ordenado

